client/blocks/signup-form/index.jsx

```diff
@@ -122,8 +123,13 @@ class SignupForm extends Component {
 
 	state = {
 		submitting: false,
-		focusPassword: false,
-		focusUsername: false,
+		isFieldDirty: {
+			email: false,
+			username: false,
+			password: false,
+			firstname: false,
+			lastnmae: false,
```

Comment:
Just out of curiosity, do you know how can one enable these fields in the form? Or in which instances they appear?

We have a typo on `lastnmae`.

Reply:
It's good you ask this question because I admit I did not test these fields when I added them to the `isFieldDirty` state object (obviously, seeing I had a typo too!). 🤦 
You enable these fields by adding the `displayNameInput={ true }` prop to this component.
Although this prop is available to the component, I don't see that we are using/enabling this prop anywhere in Calypso at this time.
I just now tested by adding the prop and found that these 2 name fields need to be camel case. (they come in from the api as snake case (`first_name`) then converted to camel case (`firstName`).  It should be all good now. 👍 Thanks for the thorough review @rcanepa ! 🙂

Reply:
Awesome. I'm glad I asked then 🙂